### PR TITLE
Fix Phabricator script

### PR DESCRIPTION
**Pull request by ijsnow**
This PR fixes the in page Phabricator script by:

1. Modifying extension host code to not rely on browser extension APIs.
2. Removing backwards compatibility support for LSP requests.
3. Fixing Diffusion `CodeView` implementation
4. Adding revision `CodeView` implementation

Closes https://github.com/sourcegraph/sourcegraph/issues/1411.

### CODE REVIEW

**Comment by ijsnow** [comment]
@chrismwendt I can't get the go extension to work with this PR. I think it is due to usage of `backcompatQueryLSP` which was removed [per this convo.](https://sourcegraph.slack.com/archives/CCLF4R6EM/p1545862999028200)

File: client/browser/src/platform/settings.ts [hunk 1]
```diff
@@ -13,32 +13,35 @@ import {
...
-                } as SettingsSubject,
-                settings: clientSettings,
-                lastID: null,
-            },
-        ],
-        final: clientSettings || {},
-    }))
-)
+export const storageSettingsCascade: Observable<SettingsCascade> = isInPage
+    ? of({ subjects: [], final: {} })
```
Comment (chrismwendt) [inline client/browser/src/platform/settings.ts]: Maybe this is an acceptable limitation, but does this line mean users won't be able to change settings (e.g. toggle Codecov line background colors)?

File: client/browser/src/shared/backend/lsp.tsx [hunk 2]
```diff
@@ -28,197 +15,15 @@ export interface LSPRequest {
...
-    if (!canFetchForURL(url)) {
-        return of([])
-    }
-    return request(url, 'textDocument/definition', body).pipe(extractLSPResponse)
-}, makeRepoURI)
-
 interface SimpleProviderFns {
     getHover: (pos: AbsoluteRepoFilePosition) => Observable<HoverMerged | null>
     fetchDefinition: (pos: AbsoluteRepoFilePosition) => Observable<Location | Location[] | null>
 }
```
Comment (chrismwendt) [inline client/browser/src/shared/backend/lsp.tsx]: I think this `SimpleProviderFns` interface is a redundant abstraction now, and can be removed. Its only purpose was to be a common interface that was implemented by both the lsp-proxy and extension implementations. Now, only the extension implementation is left.

File: shared/src/api/client/services/extensionsService.ts [hunk 3]
```diff
@@ -74,7 +74,7 @@ export class ExtensionsService {
         // Extensions that have been activated (including extensions with zero "activationEvents" that evaluate to
         // true currently).
         const activatedExtensionIDs: string[] = []
-        return combineLatest(from(this.model), this.enabledExtensions).pipe(
+        return combineLatest(from(this.model).pipe(), from(this.enabledExtensions).pipe()).pipe(
```
Comment (chrismwendt) [inline shared/src/api/client/services/extensionsService.ts]: What does `.pipe()` do? It looks like a noop to me.

File: client/browser/src/platform/settings.ts (hunk 1 above)
Comment (ijsnow) [reply client/browser/src/platform/settings.ts]: I believe that those will be synced with the user's settings on their Sourcegraph instance.

File: shared/src/api/client/services/extensionsService.ts (hunk 3 above)
Comment (ijsnow) [reply shared/src/api/client/services/extensionsService.ts]: Left in from debugging. Thanks for pointing out!

File: client/browser/src/platform/settings.ts (hunk 1 above)
Comment (sqs) [reply client/browser/src/platform/settings.ts]: Did you address this? We use client settings stored in the browser for anonymous users. @chrismwendt @ijsnow

Comment (ijsnow) [reply client/browser/src/platform/settings.ts]: `storage` is a wrapper over the browser extensions API.

**Pull request merged**